### PR TITLE
gnuplot: move gd to recommended

### PR DESCRIPTION
**Pull request by equal-l2**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Also remove some dependencies which are not used by gnuplot directly, but used by gd.

### CODE REVIEW

**Comment by ilovezfs** [comment]
Why is this a useful change?

**Comment by equal-l2** [comment]
This was useful at least just for me because I never used terminals provided by gd.

I think cairo provides better support for png than gd, and I don't use gif and jpeg, which are provided by gd.

**Comment by equal-l2** [comment]
I think the removal of dependencies is a bit more useful, since gnuplot itself doesn't need them, and `webp`, `tiff` are not used at all (no corresponding terminals).

File: Formula/gnuplot.rb [hunk 1]
```diff
@@ -82,6 +77,10 @@ def install
 
     args << "--with-pdf=#{pdflib}" if build.with? "pdflib-lite"
 
+    if build.without? "gd"
+      args << "--without-gd"
```
Comment (ilovezfs) [inline Formula/gnuplot.rb]: this can go on one line
```
args << "--without-gd" if build.without? "gd"
```

**Review by ilovezfs — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**